dynamic/msgregistry/message_registry.go

```diff
@@ -22,6 +22,42 @@ import (
 
 const googleApisDomain = "type.googleapis.com"
 
+// ErrUnexpectedEnum is returned if the URL that was requested
+// resolved to an enum instead of a message.
+type ErrUnexpectedEnum struct {
```

Comment:
How about `ErrUnexpectedType`, instead of two nearly-identical error types. We can then also add another field `ShouldBeEnum bool`. WDYT?

Reply:
Sounds good to me!